### PR TITLE
[sdk-testgen] change sdk-testgen manager from rushjs to npm

### PR DESCRIPTION
**Pull request by JackTn**
### Why move to npm from rushjs ?
The dependabot not support rushjs project. For the nodejs project, the [dependabot](https://github.com/Azure/azure-sdk-tools/actions/workflows/dependabot/dependabot-updates) will check package.json from root folder.

### CODE REVIEW

**Review by tadelesh — APPROVED** [review]

**Review by raych1 — APPROVED** [review]

File: eng/pipelines/templates/steps/sdk-testgen-set-env.yml [hunk 1]
```diff
@@ -9,7 +9,6 @@ steps:
       version: '1.22.1'
     displayName: "Install Go"
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/sdk-testgen-set-env.yml:10]: Unless this template is reused then we should probably eliminate this entire step template and move the pieces inline to the ci.yml.

File: tools/sdk-testgen/ci.yml [hunk 2]
```diff
@@ -101,15 +103,15 @@ stages:
           steps:
           - template: ../../eng/pipelines/templates/steps/sdk-testgen-set-env.yml
```
Comment (weshaggard) [inline tools/sdk-testgen/ci.yml:104]: Release should use the common JS release template, https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/stages/archetype-sdk-publish-js.yml

File: eng/pipelines/templates/steps/sdk-testgen-set-env.yml (hunk 1 above)
Comment (raych1) [reply eng/pipelines/templates/steps/sdk-testgen-set-env.yml]: Removed.

File: tools/sdk-testgen/ci.yml (hunk 2 above)
Comment (raych1) [reply tools/sdk-testgen/ci.yml]: Updated to use the common pipeline template.

File: tools/sdk-testgen/.npmrc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+registry=https://registry.npmjs.org/
```
Comment (weshaggard) [inline tools/sdk-testgen/.npmrc:1]: Why is this needed? I wouldn't expect it to be necessary.

**Review by weshaggard — APPROVED** [review]

File: tools/sdk-testgen/.npmrc (hunk 3 above)
Comment (raych1) [reply tools/sdk-testgen/.npmrc]: Removed it.
